### DISCUSSION TITLE
An error I made when defining enums/generated as union types

### DISCUSSION THREAD

**Discussion by goto40** [eclipse-langium/langium#720] [Show and tell]
Use:

    MyEnum returns string: 'on'|'off';

This returns `'on'|'off'`.

Don't:

    MyEnum returns string: ON|OFF;
    ON returns string: 'on';  // define 'on', used at different location in my grammar
    OFF returns string: 'off'; // define 'off', used at different location in my grammar

Here MyEnum returns `string`.

Reason for the first: type safety. In the second case a `string` is returned by `MyEnum`. In the first it is `'on'|'off'`.

**Comment by spoenemann** [comment]
This could be improved in the type inference. The second variant could be generated as
```typescript
export type MyEnum = ON | OFF
export type ON = 'on'
export type OFF = 'off'
```
